_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+---
+layout: post
+title:  "WildFly Glow, an evolution of WildFly Provisioning"
+date:   2024-01-10
+tags: galleon provisioning wildfly-glow
+author: jdenise
+description: WildFly Glow, an evolution of WildFly Provisioning.
+---
+
+We are here introducing an evolution of the WildFly Provisioning by the mean of the https://github.com/wildfly/wildfly-glow[WildFly Glow] project.
+
+== What is WildFly provisioning?
+
+Even if WildFly provisioning has been introduced since some time now, a quick summary seems useful to put WildFly Glow in context.
+
+WildFly Provisioning is:
+
+* The ability to create a WildFly server installation on the fly.
+* The ability to choose the set of features you want to see present in the created server.
+* The ability to extend the capabilities of WildFly server (e.g.: Datasources, keycloak SAML).
+
+The produced server has a smaller size than a full installation and the server process has a smaller memory footprint. Trimming comes with some befits:
+
+* Reduced resource consumption.
+* Smaller attack surface.
+* Simpler server configuration.
+
+https://docs.wildfly.org/galleon/[Galleon] is the technology used by WildFly provisioning.
+
+=== How provisioning is operated?
+
+From two kind of tooling:
+
+* Command Lines: Galleon CLI.
+* Maven Plugins: https://docs.wildfly.org/galleon/[Galleon Maven Plugin], https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin], https://github.com/wildfly-extras/wildfly-jar-maven-plugin[WildFly Bootable JAR Maven Plugin].
+
+=== What are the main concepts of WildFly provisioning.
+
+* Galleon feature-packs: contains metadata on how to build a WildFly server.
+* Galleon layers: contains metadata of a server feature (eg: jaxrs, ejb, jsf, ...).
+
+If you want to deep dive into Galleon concepts, you can have a look at its https://docs.wildfly.org/galleon/[documentation]. In the context of WildFly Glow, feature-packs and layers are really 
+all you need to know about.
+
+Provisioning a WildFly server is mainly assembling Galleon feature-packs and selecting Galleon layers.
+
+=== WildFly Provisioning diagram
+
+image::wildfly-glow1.jpg[WildFly provisioning diagram]
+
+This was a trimmed recap of what provisioning is. Let's see at the complexity that comes with such flexibility.
```

Comment:
The above shows a simplified representation of what provisioning is. Let's look at the....